### PR TITLE
Replace Magic Meter Items

### PR DESCRIPTION
**Pull request by Maplesstar**
Replaces both the base and double magic items with a bar. Color matches cosmetics.
Doing this to free up the regular magic models for randomizing magic refills as junk items in the pool in the future.
Chose to do the same model for both base and double magic because it's a bar, getting a second and sticking it to the first makes sense to me and making a longer one isn't really feasible without allowing it to poke out of it's space, like the box in Graveyard.

image
image
image

### CODE REVIEW

**Comment by Natheirean** [comment]
This is just a personal opinion from a runner's perspective, but I don't like the idea of this being the new magic model for magic/double magic. At the very least I would prefer an option to keep the older models in use. The model just doesn't seem all that appealing to me compared to the original models

**Comment by Maplesstar** [comment]
> This is just a personal opinion from a runner's perspective, but I don't like the idea of this being the new magic model for magic/double magic. At the very least I would prefer an option to keep the older models in use. The model just doesn't seem all that appealing to me compared to the original models

My goal is to allow those models to be used for their original purpose of magic refills, which we could then add as additional junk items. I am happy to take criticism on the models to replace them, but I would choose to remove this PR over making it an option since it would prevent making magic refills an item and defeat the entire point.

**Comment by Natheirean** [comment]
> > This is just a personal opinion from a runner's perspective, but I don't like the idea of this being the new magic model for magic/double magic. At the very least I would prefer an option to keep the older models in use. The model just doesn't seem all that appealing to me compared to the original models
> 
> My goal is to allow those models to be used for their original purpose of magic refills, which we could then add as additional junk items. I am happy to take criticism on the models to replace them, but I would choose to remove this PR over making it an option since it would prevent making magic refills an item and defeat the entire point.

Ya, I understand the purpose of it. And maybe its just me being resistant to change, but imagining seeing that as the new magic meter just feels weird to me. And I imagine for people not following the changes for this, that the new model will throw a lot of people off when it gets merged into main dev. Like it would probably warrant an announcement.

**Comment by Maplesstar** [comment]
We have the dev changelog which shows changes. I personally believe dev changes don't warrant special announcements. Especially because this is a purely visual change. We've never had announcements for major functional changes, so I see no reason to do this one special. And once a new stable releases, there's the big changelog available for that.

**Comment by SlyryD** [comment]
Could we instead put a clear tetrahedron around a magic jar (similar to freestanding dins, fw, nl -- <https://static.wikia.nocookie.net/zelda_gamepedia_en/images/a/a8/OoT_Din%27s_Fire_Model.png/revision/latest?cb=20130312011354>) to indicate the magic meter?

UPDATE: Looks like there's a dlist for that: <https://github.com/zeldaret/oot/blob/8913c4fa0d2cec19db6e8b4d8598125977243c5d/assets/xml/objects/object_gi_goddess.xml#L4>. I wonder if that can be loaded easily.

**Comment by Maplesstar** [comment]
> Could we instead put a clear tetrahedron around a magic jar (similar to freestanding dins, fw, nl -- https://static.wikia.nocookie.net/zelda_gamepedia_en/images/a/a8/OoT_Din%27s_Fire_Model.png/revision/latest?cb=20130312011354) to indicate the magic meter?
> 
> UPDATE: Looks like there's a dlist for that: https://github.com/zeldaret/oot/blob/8913c4fa0d2cec19db6e8b4d8598125977243c5d/assets/xml/objects/object_gi_goddess.xml#L4. I wonder if that can be loaded easily.

I do think this idea is worth exploring to have a comparison. Perhaps shrink the size of the magic jar slightly to give space for the semi transparent shape. Then if majority opinion is that the bar is not the best option, I will remove this PR. Unfortunately I don't know how to call portions of existing dlists in order to do try it myself.

**Comment by SlyryD** [comment]
> > Could we instead put a clear tetrahedron around a magic jar (similar to freestanding dins, fw, nl -- https://static.wikia.nocookie.net/zelda_gamepedia_en/images/a/a8/OoT_Din%27s_Fire_Model.png/revision/latest?cb=20130312011354) to indicate the magic meter?
> > UPDATE: Looks like there's a dlist for that: https://github.com/zeldaret/oot/blob/8913c4fa0d2cec19db6e8b4d8598125977243c5d/assets/xml/objects/object_gi_goddess.xml#L4. I wonder if that can be loaded easily.
> 
> I do think this idea is worth exploring to have a comparison. Perhaps shrink the size of the magic jar slightly to give space for the semi transparent shape. Then if majority opinion is that the bar is not the best option, I will remove this PR. Unfortunately I don't know how to call portions of existing dlists in order to do try it myself.

I'll give it a try -- I'm working on adding magicjars back in (<https://github.com/SlyryD/OoT-Randomizer/commit/ba9e2a21268ae913543c136b03a1d84d79399f94>). I think it's not trivial because we'll need to load the object_gi_goddess file to access the dlist. But that might just be a false understanding.

UPDATE: I managed to load the object file, put the dlist in segment 09, and then reference that dlist. Unfortunately, the dlist assumes the vertices are in segment 06, so it doesn't work properly. I'll see what I can do about that.

UPDATE: Getting somewhere 
image

**Comment by cjohnson57** [comment]
I'm still worried that sly's model looks a bit too similar to the spells, Maple's can't really be mistaken for anything else lol

File: ASM/c/item_draw_table.c [hunk 1]
```diff
@@ -163,6 +163,9 @@ item_draw_table_entry_t item_draw_table[] = {
     [0x91] = { draw_gi_c_button_vertical,   { 0x06000960, 0xFFA000FF, 1 } }, // Cdown button
     [0x92] = { draw_gi_c_button_horizontal, { 0x06000E10, 0xFFA000FF, 1 } }, // Cleft button
     [0x93] = { draw_gi_c_button_horizontal, { 0x06000E10, 0xFFA000FF, 0 } }, // Cright button
+    // [0x94] = { draw_gi_magic_meter,         { 0x060003D0, 0x06000910, 0x0600EC0, 0x00CE00FF, 0xFFFFFFFF, 0xFFFFFFFF } }, // Magic Meter , 0x000000FF -- 0x06000F70, 0x00CE00FF, --, 0x06009C0, 0x06001030, 0x00CE00FF, 0xFFFFFFFF
+    // [0x94] = { draw_gi_magic_meter,         { 0x060009F0, 0x06000ED0, 0x06001410, 0x00CE00FF, 0xFFFFFFFF, 0x000000FF } }, // Magic Meter , 0x000000FF -- 0x06000F70, 0x00CE00FF, --, 0x06009C0, 0x06001030, 0x00CE00FF, 0xFFFFFFFF
+    [0x94] = { draw_gi_magic_meter,         { 0x06000A80, 0x06000F80, 0x060014C0, 0x00CE00FF, 0xFFFFFFFF, 0x000000FF } }, // Magic Meter , 0x000000FF -- 0x06000F70, 0x00CE00FF, --, 0x06009C0, 0x06001030, 0x00CE00FF, 0xFFFFFFFF
```
Comment (fenhl) [inline ASM/c/item_draw_table.c]: Left in commented-out code.

```suggestion
    [0x94] = { draw_gi_magic_meter,         { 0x06000A80, 0x06000F80, 0x060014C0, 0x00CE00FF, 0xFFFFFFFF, 0x000000FF } }, // Magic Meter , 0x000000FF -- 0x06000F70, 0x00CE00FF, --, 0x06009C0, 0x06001030, 0x00CE00FF, 0xFFFFFFFF
```

**Review by fenhl — CHANGES_REQUESTED** [review]

**Review by fenhl — APPROVED** [review]

**Pull request merged**